components/dashboard/sections/collectives/CollectiveDetails.tsx

```diff
@@ -462,6 +464,8 @@ const CollectiveDetails = ({
   const isHostedCollective = host && collective?.host?.id === host?.id;
   const isLoading = loading || loadingCollectiveInfo;
   const isChild = !!collective?.parent?.id;
+  const lastUrl = history[history.length - 2];
+  const backToUrl = /\/dashboard\/[\w\d-]+\/(hosted|all)-collectives\/[\w\d-]{35}/i.test(lastUrl) ? lastUrl : null;
```

Comment:
Testing string URLs with a regex is always a bit risky (see ____). A safer approach is to first parse it:

```ts
const parsedUrl = new URL(url); // TODO: Handle parse exceptions
const regex = /^{regexContent}$/; // Regex with start + end
return regex.test(parsedUrl.pathname);
```